aioh2/protocol.py

```diff
@@ -362,6 +365,23 @@ def _priority_step(self):
 
     # APIs
 
+    def set_handler(self, handler):
+        """
+        Connect with a coroutine, which is called when connection is made.
```

Comment:
It's true when the task has an exit point (return or exception), but for an infinite loop (like WebSocket), when the connection is terminated unexpectedly, it'll just hang there. So it's better to explicitly terminate the task.
I don't know but I'll take a look at how `asyncio` does that (or does it assume the task to end itself?)

I think it's better the task itself handles the cancel situation (by catch CancelledError, and it can do cleanup there), and if it catches and decides not to end, that's the user's choice.